### PR TITLE
[@types/react-native] touchSoundDisabled TouchableWithoutFeedback prop

### PR DESCRIPTION
**Pull request by alexeymolchan**
Added touchSoundDisabled prop for TouchableWithoutFeedback component

Please fill in this template.

Use a meaningful title for the pull request. Include the name of the package modified.
Test the change in your own code. (Compile and run.)
Add or edit tests to reflect the change. (Run with npm test.)
Follow the advice from the readme.
Avoid common mistakes.
Run npm run lint package-name (or tsc if no tslint.json is present).

prop docs link: https://reactnative.dev/docs/touchablewithoutfeedback#touchsounddisabled

flow src link: https://github.com/facebook/react-native/blob/master/Libraries/Components/Touchable/TouchableWithoutFeedback.js#L68

### CODE REVIEW

**Comment by alloy** [comment]
@alexeymolchan Please be sure to provide links to the docs and/or Flow types in the upstream code-base, as per the instructions in the PR template 🙏

**Comment by alexeymolchan** [comment]
> @alexeymolchan Please be sure to provide links to the docs and/or Flow types in the upstream code-base, as per the instructions in the PR template 🙏

done

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -4974,10 +4975,19 @@ export interface TouchableWithoutFeedbackPropsIOS {
     tvParallaxProperties?: TVParallaxProperties;
 }
 
+export interface TouchableWithoutFeedbackPropsAndroid {
```
Comment (alloy) [inline types/react-native/index.d.ts:4978]: 👍

File: types/react-native/index.d.ts [hunk 2]
```diff
@@ -4974,10 +4975,19 @@ export interface TouchableWithoutFeedbackPropsIOS {
...
+    /**
+     * If true, doesn't play a system sound on touch.
+     *
+     * @platform android
+     */
+    touchSoundDisabled?: boolean;
```
Comment (alloy) [inline types/react-native/index.d.ts:4984]: Technically the upstream Flow typing is:

```suggestion
    touchSoundDisabled?: boolean | null;
```

**Review by alloy — CHANGES_REQUESTED** [review]
Thanks! Just one minor change suggestion.

**Review by alexeymolchan — COMMENTED** [review]
> Thanks! Just one minor change suggestion.

@alloy so ?

File: types/react-native/index.d.ts (hunk 2 above)
Comment (alloy) [reply types/react-native/index.d.ts:4984]: A Flow type prefixed with a question mark is a “maybe” type. Both `undefined` and `null` are assignable to Flow’s “maybe” types: https://flow.org/en/docs/types/maybe/

Comment (alexeymolchan) [reply types/react-native/index.d.ts:4984]: @allow i see but disabled prop has the same prefixed question mark. So should we change it as well then ?

Comment (alloy) [reply types/react-native/index.d.ts:4984]: > @Allow i see but disabled prop has the same prefixed question mark. So should we change it as well then ?

Yup, please do 👍 

To be honest, there are probably more cases that should be covered, but for now I mostly want to make sure we don’t introduce any new cases.

Comment (alexeymolchan) [reply types/react-native/index.d.ts]: @alloy done

**Review by alloy — APPROVED** [review]

File: types/react-native/index.d.ts (hunk 2 above)
Comment (alloy) [reply types/react-native/index.d.ts]: Great, thanks!

**Pull request merged**